### PR TITLE
Updates for vector-read update.

### PR DESCRIPTION
**Pull request by LetThereBeDwight**
Check for consistency given the previous coupling of layers and featuresets. Related to DigialGlobe/DeepCore-Dev#329 DigialGlobe/DeepCore-Dev#371 and DigialGlobe/DeepCore-Dev#400. Should not be merged until those are complete.

### CODE REVIEW

File: src/OpenSpaceNet.cpp [hunk 1]
```diff
@@ -53,6 +53,8 @@
 #include <utility/Semaphore.h>
 #include <utility/User.h>
 
+#include <ogr_spatialref.h>
```
Comment (avitebskiy) [inline src/OpenSpaceNet.cpp]: remove

File: src/OpenSpaceNet.cpp [hunk 2]
```diff
@@ -275,7 +277,11 @@ void OpenSpaceNet::initFeatureSet()
 
     VectorOpenMode openMode = args_.append ? APPEND : OVERWRITE;
 
-    featureSet_ = make_unique<FeatureSet>(args_.outputPath, args_.outputFormat, args_.layerName, definitions, openMode);
+    featureSet_ = make_unique<FeatureSet>(args_.outputPath, args_.outputFormat, openMode);
```
Comment (avitebskiy) [inline src/OpenSpaceNet.cpp]: change lines 281-284 to 
```cpp
featureSet_->createLayer(args_.layerName, SpatialReference::WGS84, args_.geometryType, definitions);
```

File: src/OpenSpaceNet.cpp [hunk 3]
```diff
@@ -450,7 +456,49 @@ void OpenSpaceNet::processSerial()
     }
 }
 
-void OpenSpaceNet::addFeature(const cv::Rect &window, const vector<Prediction> &predictions)
+void OpenSpaceNet::addFeature(const cv::Rect &window, const vector<Prediction> &predictions, const string& layerName)
```
Comment (avitebskiy) [inline src/OpenSpaceNet.cpp]: Should be
```cpp
void OpenSpaceNet::addFeature(const cv::Rect &window, const vector<Prediction> &predictions)
```
use the cached `Layer`

File: src/OpenSpaceNet.cpp [hunk 4]
```diff
@@ -450,7 +456,49 @@ void OpenSpaceNet::processSerial()
...
+                                                     move(createFeatureFields(predictions))));
+        }
+            break;
+
+        default:
+            DG_ERROR_THROW("Invalid output type");
+    }
+}
+
+void OpenSpaceNet::addFeature(const cv::Rect &window, const vector<Prediction> &predictions, int index)
```
Comment (avitebskiy) [inline src/OpenSpaceNet.cpp]: Remove, shouldn't be used.

File: src/OpenSpaceNet.cpp (hunk 2 above)
Comment (LetThereBeDwight) [inline src/OpenSpaceNet.cpp]: I'm assuming you mean 278-282

**Comment by avitebskiy** [comment]
test this please

**Pull request merged**